src/db_cache/mod.rs

```diff
@@ -124,9 +128,9 @@ pub const DEFAULT_MAX_CAPACITY: u64 = 64 * 1024 * 1024;
 /// ```
 #[async_trait]
 pub trait DbCache: Send + Sync {
-    async fn get_block(&self, key: CachedKey) -> Option<CachedEntry>;
-    async fn get_index(&self, key: CachedKey) -> Option<CachedEntry>;
-    async fn get_filter(&self, key: CachedKey) -> Option<CachedEntry>;
+    async fn get_block(&self, key: CachedKey) -> Result<Option<CachedEntry>, SlateDBError>;
+    async fn get_index(&self, key: CachedKey) -> Result<Option<CachedEntry>, SlateDBError>;
+    async fn get_filter(&self, key: CachedKey) -> Result<Option<CachedEntry>, SlateDBError>;
```

Comment:
it seems all the calling sites all treat the errors as 'cache not found', what do you think about not changing this function signature by return `None` with an warning log until we found an recommended way for handling errors for the callers during db cache access? iiuc we can consider this as the error has already been handled in the db cache impl, so the caller do not require to further handle this error again.

Reply:
oh ignore me, this error is handled in db cacher wrapper, and db cache wrapper requires this `Result` from the impl to handle this.